<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>WakeSPRK</class>
 <widget class="QMainWindow" name="WakeSPRK">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>702</width>
    <height>622</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>WakeSPRK</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_4">
    <item>
     <layout class="QHBoxLayout" name="visualLayout">
      <item>
       <widget class="QOpenGLWidget" name="openGLWidget">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QTabWidget" name="tabWidget">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="currentIndex">
         <number>1</number>
        </property>
        <widget class="QWidget" name="tab">
         <attribute name="title">
          <string>Manual</string>
         </attribute>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout" stretch="0">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout">
              <item>
               <layout class="QFormLayout" name="formLayout_3">
                <item row="0" column="0">
                 <widget class="QLabel" name="label_4">
                  <property name="text">
                   <string>X Rotation</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <widget class="QDial" name="xDial">
                  <property name="minimum">
                   <number>-30</number>
                  </property>
                  <property name="maximum">
                   <number>30</number>
                  </property>
                  <property name="notchesVisible">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
                <item row="1" column="0">
                 <widget class="QLabel" name="label_5">
                  <property name="text">
                   <string>Y Rotation</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="1">
                 <widget class="QDial" name="yDial">
                  <property name="minimum">
                   <number>-30</number>
                  </property>
                  <property name="maximum">
                   <number>30</number>
                  </property>
                  <property name="notchesVisible">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
                <item row="2" column="0">
                 <widget class="QLabel" name="label_6">
                  <property name="text">
                   <string>Z Rotation</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="1">
                 <widget class="QDial" name="zDial">
                  <property name="minimum">
                   <number>-30</number>
                  </property>
                  <property name="maximum">
                   <number>30</number>
                  </property>
                  <property name="notchesVisible">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QFormLayout" name="formLayout">
                <property name="leftMargin">
                 <number>5</number>
                </property>
                <item row="0" column="0">
                 <widget class="QLabel" name="label">
                  <property name="text">
                   <string>X Position</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <widget class="QSlider" name="xSlider">
                  <property name="minimum">
                   <number>-50</number>
                  </property>
                  <property name="maximum">
                   <number>50</number>
                  </property>
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                 </widget>
                </item>
                <item row="1" column="0">
                 <widget class="QLabel" name="label_2">
                  <property name="text">
                   <string>Y Position</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="1">
                 <widget class="QSlider" name="ySlider">
                  <property name="minimum">
                   <number>-50</number>
                  </property>
                  <property name="maximum">
                   <number>50</number>
                  </property>
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                 </widget>
                </item>
                <item row="2" column="0">
                 <widget class="QLabel" name="label_3">
                  <property name="text">
                   <string>Z Position</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="1">
                 <widget class="QSlider" name="zSlider">
                  <property name="minimum">
                   <number>-50</number>
                  </property>
                  <property name="maximum">
                   <number>50</number>
                  </property>
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <widget class="QPushButton" name="homeBtn">
                <property name="text">
                 <string>Home</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="runTrajectoryBtn">
                <property name="text">
                 <string>Run Trajectory</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="verticalSpacer">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>40</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab_2">
         <attribute name="title">
          <string>Setup</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <item>
           <widget class="QGroupBox" name="groupBox">
            <property name="title">
             <string>Communication</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_7">
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_7">
               <item>
                <widget class="QLabel" name="label_13">
                 <property name="text">
                  <string>Port Name</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="portNameLine">
                 <property name="text">
                  <string>COM5</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QPushButton" name="enableBtn">
               <property name="text">
                <string>Enable Motors</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox_2">
            <property name="title">
             <string>Send Message</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_8">
             <item>
              <layout class="QFormLayout" name="formLayout_4">
               <item row="0" column="0">
                <widget class="QLabel" name="label_7">
                 <property name="text">
                  <string>Address</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <layout class="QHBoxLayout" name="horizontalLayout_12">
                 <item>
                  <widget class="QLineEdit" name="addrLine">
                   <property name="sizePolicy">
                    <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="text">
                    <string>84</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QComboBox" name="addrCBox">
                   <property name="editable">
                    <bool>false</bool>
                   </property>
                   <item>
                    <property name="text">
                     <string>P Gain</string>
                    </property>
                   </item>
                   <item>
                    <property name="text">
                     <string>I Gain</string>
                    </property>
                   </item>
                   <item>
                    <property name="text">
                     <string>D Gain</string>
                    </property>
                   </item>
                   <item>
                    <property name="text">
                     <string>V Trajectory</string>
                    </property>
                   </item>
                   <item>
                    <property name="text">
                     <string>A Trajectory</string>
                    </property>
                   </item>
                  </widget>
                 </item>
                </layout>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="label_8">
                 <property name="text">
                  <string>Data</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <layout class="QHBoxLayout" name="horizontalLayout_13">
                 <item>
                  <widget class="QLineEdit" name="dataLine">
                   <property name="sizePolicy">
                    <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QPushButton" name="sendPacket">
                   <property name="text">
                    <string>Send</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox_3">
            <property name="title">
             <string>Sine Trajectory</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_9">
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_3">
               <item>
                <widget class="QCheckBox" name="sinX">
                 <property name="text">
                  <string>X Position</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                 <attribute name="buttonGroup">
                  <string notr="true">sinBtnGrp</string>
                 </attribute>
                </widget>
               </item>
               <item>
                <widget class="QCheckBox" name="sinY">
                 <property name="text">
                  <string>Y Position</string>
                 </property>
                 <attribute name="buttonGroup">
                  <string notr="true">sinBtnGrp</string>
                 </attribute>
                </widget>
               </item>
               <item>
                <widget class="QCheckBox" name="sinZ">
                 <property name="text">
                  <string>Z Position</string>
                 </property>
                 <attribute name="buttonGroup">
                  <string notr="true">sinBtnGrp</string>
                 </attribute>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_5">
               <item>
                <widget class="QCheckBox" name="sinRx">
                 <property name="text">
                  <string>X Rotation</string>
                 </property>
                 <attribute name="buttonGroup">
                  <string notr="true">sinBtnGrp</string>
                 </attribute>
                </widget>
               </item>
               <item>
                <widget class="QCheckBox" name="sinRy">
                 <property name="text">
                  <string>Y Rotation</string>
                 </property>
                 <attribute name="buttonGroup">
                  <string notr="true">sinBtnGrp</string>
                 </attribute>
                </widget>
               </item>
               <item>
                <widget class="QCheckBox" name="sinRz">
                 <property name="text">
                  <string>Z Rotation</string>
                 </property>
                 <attribute name="buttonGroup">
                  <string notr="true">sinBtnGrp</string>
                 </attribute>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QFormLayout" name="formLayout_5">
               <item row="1" column="0">
                <widget class="QLabel" name="label_9">
                 <property name="text">
                  <string>Amplitude</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QLineEdit" name="ampLine">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QLabel" name="label_10">
                 <property name="text">
                  <string>Period</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="1">
                <widget class="QLineEdit" name="periodLine">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                </widget>
               </item>
               <item row="3" column="1">
                <widget class="QPushButton" name="runSineBtn">
                 <property name="text">
                  <string>Run</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>171</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="sinBtnGrp"/>
 </buttongroups>
</ui>
